### ISSUE TITLE
CPLEX.jl 0.3.3 installed at the same time as JuMP 0.20

### ISSUE THREAD

**Issue by dourouc05** [JuliaOpt/CPLEX.jl#255]
I've just installed Julia 1.3 RC 2, and I was installing my packages when something bad happened. I used this line to install my bunch of packages: 

`add JuMP CPLEX LightGraphs Cairo MathOptFormat MetaGraphs Compose Colors GraphPlot Fontconfig`

Then, Pkg installed... CPLEX 0.3.3, along with JuMP 0.20. 

Of course, when importing things, it all crashes very quickly: 

```
julia> using CPLEX
[ Info: Precompiling CPLEX [a076750e-1247-5638-91d2-ce28b192dca0]
ERROR: LoadError: syntax: extra token "MathProgBase" after end of expression
Stacktrace:
 [1] top-level scope at C:\Users\Thibaut\.julia\packages\CPLEX\FUtHu\src\CPLEX.jl:23
 [2] include at .\boot.jl:328 [inlined]
 [3] include_relative(::Module, ::String) at .\loading.jl:1105
 [4] include(::Module, ::String) at .\Base.jl:31
 [5] top-level scope at none:2
 [6] eval at .\boot.jl:330 [inlined]
 [7] eval(::Expr) at .\client.jl:433
 [8] top-level scope at .\none:3
in expression starting at C:\Users\Thibaut\.julia\packages\CPLEX\FUtHu\src\CPLEX.jl:23
ERROR: Failed to precompile CPLEX [a076750e-1247-5638-91d2-ce28b192dca0] to C:\Users\Thibaut\.julia\compiled\v1.3\CPLEX\QpT0w_MeaLm.ji.
Stacktrace:
 [1] error(::String) at .\error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at .\loading.jl:1283
 [3] _require(::Base.PkgId) at .\loading.jl:1024
 [4] require(::Base.PkgId) at .\loading.jl:922
 [5] require(::Module, ::Symbol) at .\loading.jl:917
```

I suppose that something is not configured properly?

**Comment by odow** [comment]
CPLEX has not been updated to MOI 0.9, so `Pkg` can either install the latest JuMP and an old CPLEX which uses `MathProgBase` instead of MOI, or an old JuMP and the latest CPLEX. It chooses the latest JuMP.

**Comment by dourouc05** [comment]
For now, it installs a combination that makes no sense: the latest JuMP, with a pre-MOI CPLEX; I would totally be fine with JuMP 0.19 and CPLEX 0.5, a combination that works, but that I have to specify myself. There must be a dependency that is not properly configured somewhere.

**Comment by odow** [comment]
The issue is that there was no upper bound for Julia in the `REQUIRE`:
https://github.com/JuliaOpt/CPLEX.jl/blob/v0.3.3/REQUIRE

But the following line looks like it restricts Julia to <1.0
https://github.com/JuliaRegistries/General/blob/master/C/CPLEX/Compat.toml#L47

I'm not really sure what the problem is.

**Comment by blegat** [comment]
`julia = "0.6-1"` includes Julia v1, to exclude it it should be `julia = "0.6-0"` I think. The syntax of this file is not documented anywhere so it's not easy to parse.

**Comment by odow** [comment]
Closed by the v0.6.0 tag: https://github.com/JuliaOpt/CPLEX.jl/releases/tag/v0.6.0.
